pkg/provider/aws/expose.go

```diff
@@ -0,0 +1,123 @@
+				pathParam = strings.ReplaceAll(pathParam, ":", "")
+				pathParam = strings.ReplaceAll(pathParam, "*", "")
+				methodRequestParams[fmt.Sprintf("method.%s", pathParam)] = true
+				integrationRequestParams[fmt.Sprintf("integration.%s", pathParam)] = fmt.Sprintf("method.%s", pathParam)
+			}
+
+			segment = convertPath(segment)
+			currPathSegment.WriteString(fmt.Sprintf("%s/", segment))
+			refs := []core.AnnotationKey{gateway.Provenance(), execUnit.Provenance()}
+			resource, ok := resourceMap[segment]
+			if !ok {
+				resource = resources.NewApiResource(api, refs, segment)
+				dag.AddResource(resource)
+				dag.AddDependency2(resource, api)
+				resourceMap[currPathSegment.String()] = resource
+				triggers[resource.Name] = resource.Name
+			}
+
+			method := resources.NewApiMethod(resource, refs, strings.ToUpper(string(route.Verb)), methodRequestParams)
+			dag.AddResource(method)
+			dag.AddDependency2(method, resource)
+			integration, err := a.createIntegration(method, execUnit, refs, route, dag)
+			if err != nil {
+				errs.Append(err)
+				continue
+			}
+
+			triggers[integration.Name] = integration.Name
+		}
+	}
+	api.ConstructsRef = api_references
+	deployment := resources.NewApiDeployment(api, api_references, triggers)
+	dag.AddResource(deployment)
+	dag.AddDependency2(deployment, api)
+	stage := resources.NewApiStage(deployment, "$default", api_references)
+	dag.AddResource(stage)
+	dag.AddDependency2(stage, deployment)
+	return errs.ErrOrNil()
+}
+
+func (a *AWS) createIntegration(method *resources.ApiMethod, unit *core.ExecutionUnit, refs []core.AnnotationKey, route core.Route, dag *core.ResourceGraph) (*resources.ApiIntegration, error) {
+	cfg := a.Config.GetExecutionUnit(unit.ID)
+	switch cfg.Type {
+	case Lambda:
+		constructResources, _ := a.GetResourcesDirectlyTiedToConstruct(unit)
+		if len(constructResources) != 1 {
+			return nil, errors.Errorf("Expected one resource to be tied to a lambda execution unit, %s, but found %s", unit.ID, strconv.Itoa(len(constructResources)))
+		}
+		function, ok := constructResources[0].(*resources.LambdaFunction)
+		if !ok {
+			return nil, errors.Errorf("Expected resource to be of type, lambda function, for execution unit, %s", unit.ID)
+		}
+		lambdaPermission := resources.NewLambdaPermission(function, core.IaCValue{Resource: method.RestApi, Property: core.ARN_IAC_VALUE}, "apigateway.amazonaws.com", "lambda:InvokeFunction", refs)
+		dag.AddResource(lambdaPermission)
+		dag.AddDependency2(lambdaPermission, function)
+		integration := resources.NewApiIntegration(method, refs, "POST", "AWS_PROXY", nil, core.IaCValue{Resource: function, Property: resources.LAMBDA_INTEGRATION_URI_IAC_VALUE})
+		dag.AddResource(integration)
+		dag.AddDependency2(integration, method)
+		dag.AddDependency2(integration, function)
+		return integration, nil
+	default:
+		return nil, errors.Errorf("Unrecognized integration type, %s, for api gateway", cfg.Type)
+	}
+}
+
+func convertPath(path string) string {
```

Comment:
convert from what to what? is this leaking a particular library's route syntax (e.g. for var substitution)?

Reply:
i believe this is specific for api gw but let me comment whats going on. Ill just add a bunch of comments

Reply:
Ah cool. If it's specific for API GW that's totally fine in `provider/aws`. I just wanted to make sure it's not specific to express, or something like that.